.github/workflows/build.yml

```diff
@@ -41,6 +41,14 @@ jobs:
       - run: npm test
         working-directory: browser
 
+      - name: Cache OPAM
+        uses: actions/cache@v3
+        with:
+          path: ~/.opam
+          key: ${{ runner.os }}-opam-${{ matrix.ocaml-compiler }}-${{ hashFiles('**/*.opam') }}
+          restore-keys: |
+            ${{ runner.os }}-opam-${{ matrix.ocaml-compiler }}-
```

Comment:
Is this ok? I would expect the `key` to be the same as the `restore-keys`

Reply:
See https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key

Reply:
This does make complete sense. Otherwise is will create a new cache for each code change.

For tidy-ness the trail `-` might not be necessary.